README.md

```diff
@@ -35,64 +43,71 @@ Create/retrieve key:
                                   UpdateString(key1, "foo value2").
                                   UpsertString(key2, "bar value").
                                   Build()
 
 
-
 To create a new tagsSet from an existing tag set oldTagSet:
+
     oldTagSet := ...
     tsb := NewTagSetBuilder(oldTagSet)
     tsb.InsertString(key1, "foo value")
     tsb.UpdateString(key1, "foo value2")
     tsb.UpsertString(key2, "bar value")
     newTagSet := tsb.Build()
 
-    // A shorter way of achieving the same is:
+The following is a shorter way of achieving the same:
+
     oldTagSet := ...
-	newTagSet := NewTagSetBuilder(oldTagSet).InsertString(key1, "foo value").
-		UpdateString(key1, "foo value2").
-		UpsertString(key2, "bar value").
-		Build()    
-
-### Add new tagSet to a context / Modify tagSet in a context 
-Add tags to a context for propagation to downstream methods and downstream rpcs:
-To create a new context with the tags. This will create a new context where all the existing tags in the current context are deleted and replaced with the tags passed as argument.
-    
+    newTagSet := NewTagSetBuilder(oldTagSet).InsertString(key1, "foo value").
+                 UpdateString(key1, "foo value2").
+                 UpsertString(key2, "bar value").
+                 Build()
+
+### To add or modify tagSet in a context
+
+A tag set can be added to a context so the tags are propagated to downstream
+methods and remote procedure calls with the context.
+
+To create a new context with the tags,
+
     newTagSet  := ...
     ctx2 := tags.NewContext(ctx, newTagSet)
 
-Create a new context keeping the old tag set and adding new tags to it, removing specific tags from it, or modifying the values fo some tags. This is just a matter of getting the oldTagSet from the context, apply changes to it, then create a new  context with the newTagSet.
-    
+The above code will create a new context where all the existing tags in the current context are deleted and replaced with the tags passed as argument.
+If a new context should be created by modifying the existing tag set in the
+context, rather than creating one from scratch
+(for example, adding new tags, removing some tags, or changing the
+values of some tags), we can retrieve the old tag set from the context,
+create a TagSetBuilder based on the old tag set, and create a new context
+with the newly created tag set.
+
     oldTagSet := tags.FromContext(ctx)
     newTagSet := NewTagSetBuilder(oldTagSet).InsertString(key1, "foo value").
         UpdateString(key1, "foo value2").
         UpsertString(key2, "bar value").
         Build()
-
     ctx2 := tags.NewContext(ctx, newTagSet)
 
-## Stats
+## Stats API
 
 ### To create/retrieve/delete a measure a.k.a resource
-Create/load measures units:
+
+Create and load measures with unit:
 
     // returns a *MeasureFloat64
     mf, err := stats.NewMeasureFloat64("/my/float64/measureName", "some measure")
```

Comment:
There is a typo here (from me). This should be:
stats.NewMeasureFloat64("/my/float64/measureName", "some measure", "MBy") or  stats.NewMeasureFloat64("/my/float64/measureName", "some measure", "1")

The unit is actually just a string that obeys the rules defined here http://unitsofmeasure.org/ucum.html

Reply:
Updated. Should mention that in the stats package doc. I assume this unit is just for human consumption (otherwise, I guess it should be a separate type instead of a string)

"1" is a strange string for unit. MBy is... it's my first time to encounter with it :-)